src/Bicep.Core/SourceGraph/BicepSourceFile.cs

```diff
@@ -110,6 +110,31 @@ public ResultWithDiagnosticBuilder<AuxiliaryFile> TryLoadAuxiliaryFile(RelativeP
                 });
         }
 
+        public ResultWithDiagnosticBuilder<IEnumerable<IOUri>> TryListFilesInDirectory(RelativePath relativePath, string searchPattern = "")
+        {
+            if (this.FileHandle is DummyFileHandle)
+            {
+                // This is only invoked when building SnippetCache. The error is swallowed.
+                return new(x => x.ErrorOccurredReadingFile("Cannot load auxiliary file from dummy file handle"));
+            }
+
+            var directoryHandle = this.FileHandle.GetParent().GetDirectory(relativePath);
```

Comment:
I wonder if we could add a validation to check whether `relativePath` resolves to a file and, if so, return an error like "Found a file instead of a directory." We perform a similar check when resolving relative files (see `IFileHandleExtensions.TryResolveRelativeFile`).